### PR TITLE
Add docker dev build targets

### PR DESCRIPTION
**Pull request by jasonodonnell**
This adds a slightly modified copy of the `docker-vault` Dockerfile to the repo for building and testing dev Vault builds.  This is helpful if you want to test development changes of Vault in Docker.

### CODE REVIEW

File: scripts/docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+VOLUME /vault/file
+
+# 8200/tcp is the primary interface that applications use to interact with
+# Vault.
+EXPOSE 8200
+
+# The entry point script uses dumb-init as the top-level process to reap any
+# zombie processes created by Vault sub-processes.
+#
+# For production derivatives of this container, you shoud add the IPC_LOCK
```
Comment (kalafut) [inline scripts/docker/Dockerfile:39]:
```suggestion
# For production derivatives of this container, you should add the IPC_LOCK
```

File: Makefile [hunk 2]
```diff
@@ -52,6 +52,12 @@ dev-ui-mem: assetcheck dev-ui
 dev-dynamic-mem: BUILD_TAGS+=memprofiler
 dev-dynamic-mem: dev-dynamic
 
+# Creates a Docker image by adding the compiled linux/amd64 binary found in ./bin.
+# The resulting image is tagged "vault:dev". 
+docker-dev: prep
```
Comment (kalafut) [inline Makefile:57]: Maybe have a `docker-dev-ui` target as well, complementing `dev-ui`?

**Comment by jasonodonnell** [comment]
@catsby I added the multi-stage build

File: Makefile (hunk 2 above)
Comment (jasonodonnell) [reply Makefile:57]: Done!

**Pull request merged**